hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java

```diff
@@ -131,6 +133,45 @@ public DataSerializable read(ObjectDataInput in) throws IOException {
         }
     }
 
+    @Override
+    public DataSerializable read(ObjectDataInput in, Class aClass)
```

Comment:
This method shares most code with the existing `DataSerializable read(ObjectDataInput in)` method, it would be nice to deduplicate (eg have both public methods delegate to an internal method). The same is true for most changes under `internal/serialization` package.